### ISSUE TITLE
Error to connect kafka when I change version to v0.9.0

### ISSUE THREAD

**Issue by rosicleiafrasson** [mostafa/xk6-kafka#67]
My script worked in version v0.8.0. After I switch to version 0.9.0 I started getting this error:

Failed to write message: read tcp ip: read: connection reset by peerFailed to write message.

```
import { check } from "k6";
import {
    writer,
    produceWithConfiguration,
    createTopic,
} from "k6/x/kafka"; 
import http from 'k6/http';
import encoding from 'k6/encoding';

const bootstrapServers = ["xxx:9092"];
const topic = "payin";

const auth = JSON.stringify({
    username: "xxx",
    password: "xxx",
    algorithm: "SHA512",
});

const producer = writer(bootstrapServers, topic, auth);

var configuration = JSON.stringify({
    producer: {
        keySerializer: "io.confluent.kafka.serializers.KafkaAvroSerializer",
        valueSerializer: "io.confluent.kafka.serializers.KafkaAvroSerializer",
    },
    schemaRegistry: {
        url: "xxxx",
        basicAuth: {
            credentialsSource: "USER_INFO",
            userInfo: "xxx:xxx",
        },
    },
});

export default function () {

    const credentials = `xxx:xxx`;
    const encodedCredentials = encoding.b64encode(credentials);
    const options = {
        headers: {
        Authorization: `Basic ${encodedCredentials}`,
        },
    };

const result = http.get('xxxx', options)
const schema = JSON.parse(result.body).schema

let messages = [
    {
        value: JSON.stringify({
            "status": "PROPOSED"
        }),
    },
];

let error = produceWithConfiguration(
    producer,
    messages,
    configuration,
    null,
    schema
);

check(error, {
    "is sent": (err) => err == undefined,
});

}
```

**Comment by rosicleiafrasson** [comment]
`
import { check } from "k6";
import {
    writer,
    produceWithConfiguration,
    createTopic,
} from "k6/x/kafka"; // import kafka extension
import http from 'k6/http';
import encoding from 'k6/encoding';

const bootstrapServers = ["xxx:9092"];
const topic = "payin";

const auth = JSON.stringify({
    username: "xxx",
    password: "xxx",
    algorithm: "SHA512",
});

const producer = writer(bootstrapServers, topic, auth);

var configuration = JSON.stringify({
    producer: {
        keySerializer: "io.confluent.kafka.serializers.KafkaAvroSerializer",
        valueSerializer: "io.confluent.kafka.serializers.KafkaAvroSerializer",
    },
    schemaRegistry: {
        url: "xxxx",
        basicAuth: {
            credentialsSource: "USER_INFO",
            userInfo: "xxx:xxx",
        },
    },
});


export default function () {

    const credentials = `xxx:xxx`;
    const encodedCredentials = encoding.b64encode(credentials);
    const options = {
        headers: {
        Authorization: `Basic ${encodedCredentials}`,
        },
    };

const result = http.get('xxxx', options)
const schema = JSON.parse(result.body).schema




let messages = [
    {
        value: JSON.stringify({
            "status": "PROPOSED"
        }),
    },
];

let error = produceWithConfiguration(
    producer,
    messages,
    configuration,
    null,
    schema
);



check(error, {
    "is sent": (err) => err == undefined,
});




`

**Comment by eabreusantos** [comment]
I'm having the same behavior.

**Comment by mostafa** [comment]
@rosicleiafrasson @eabreusantos 
The APIs are slightly changed. Please check the scripts in the [v0.9.0](https://github.com/mostafa/xk6-kafka/tree/v0.9.0) tree.

**Comment by mostafa** [comment]
@rosicleiafrasson @eabreusantos 
Any updates?

**Comment by rosicleiafrasson** [comment]
Yes, I analyzed the way authentication in the v0.9.0 scripts. But the way is the same. I still have the same problem.

**Comment by mostafa** [comment]
@rosicleiafrasson @eabreusantos
Please update your scripts according to the [latest changes](https://github.com/mostafa/xk6-kafka/blob/main/docs/README.md) to the API, as shown in [this script](https://github.com/mostafa/xk6-kafka/blob/main/scripts/test_avro_with_schema_registry.js) and then [build](https://github.com/mostafa/xk6-kafka#build-for-development) the latest `main` branch and test it to see if it works.

**Comment by mostafa** [comment]
Released in [v0.11.0](https://github.com/mostafa/xk6-kafka/releases/tag/v0.11.0).
